sklearn/model_selection/_search.py

```diff
@@ -997,3 +1031,491 @@ def fit(self, X, y=None, labels=None):
+
+    error_score : 'raise' (default) or numeric
+        Value to assign to the score if an error occurs in estimator fitting.
+        If set to 'raise', the error is raised. If a numeric value is given,
+        FitFailedWarning is raised. This parameter does not affect the refit
+        step, which will always raise the error.
+
+    n_candidates : int, default=500
+        Number of random candidates to sample for each GP iteration
+        Default is 500.
+
+    n_local_candidates: int, default=10
+        After every iteration that gives us the best predicted value
+        from the GP posterior, the number of local candidates to sample
+        and add to the grid which are very close to this best candidate.
+
+    gp_params : dict, default={}
+        Parameters to pass to the GaussianProcessRegressor object.
+
+    xi : float, default=0.01
+        Exploration-exploitation trade-off parameter for the expected
+        improvement acquisition function. Higher values for this parameter
+        yield updates that are more conservative in which the exploitation
+        criterion dominates over the exploration criterion.
+
+    kappa: float, default=1.96
+        Exploration-exploitation trade-off parameter for the upper
+        confidence bound acquisition function. The default value of
+        1.96 corresponds to the 95% confidence bound on the predicted
+        objective function.
+
+    Attributes
+    ----------
+    grid_scores_ : list of named tuples
+        Contains scores for all parameter combinations in param_grid.
+        Each entry corresponds to one parameter setting.
+        Each named tuple has the attributes:
+
+            * ``parameters``, a dict of parameter settings
+            * ``mean_validation_score``, the mean score over the
+              cross-validation folds
+            * ``cv_validation_scores``, the list of scores for each fold
+
+    best_estimator_ : estimator
+        Estimator that was chosen by the search, i.e. estimator
+        which gave highest score (or smallest loss if specified)
+        on the left out data. Not available if refit=False.
+
+    best_score_ : float
+        Score of best_estimator on the left out data.
+
+    best_params_ : dict
+        Parameter setting that gave the best results on the hold out data.
+
+    Notes
+    -----
+    The parameters selected are those that maximize the score of the held-out
+    data, according to the scoring parameter.
+
+    See Also
+    --------
+    :class:`GridSearchCV`:
+        Does exhaustive search over a grid of parameters.
+
+    :class:`RandomizedSearchCV`:
+        Randomized search on hyper parameters.
+
+    Examples
+    --------
+    >>> from sklearn import linear_model, datasets
+    >>> from sklearn.model_selection import SequentialSearchCV
+    >>> from scipy import stats
+    >>> X, y = datasets.make_regression()
+    >>> clf = linear_model.Ridge()
+    >>> parameters = {"alpha": stats.expon(scale=100)}
```

Comment:
I think this contradicts the structure that you defined before on how the ```params_distribution``` argument should be before (dict of dicts).